pkg/inventory/model/journal.go

```diff
@@ -169,15 +179,33 @@ func (w *Watch) Start() {
 			w.Handler.End()
 			w.log.V(3).Info("watch stopped.")
 		}()
-		count := 0
+		for {
+			m, hasNext, err := snapshot.Next()
+			if err != nil {
+				w.log.Error(err, "next() failed.")
+				w.Handler.Error(err)
+				return
+			}
+			if hasNext {
+				w.Handler.Created(
+					Event{
+						Action: Created,
+						Model:  m.(Model),
+					})
+			} else {
+				w.log.V(3).Info("has parity.")
+				w.Handler.Parity()
+				break
+			}
+		}
 		for itr := range w.queue {
 			for {
 				event := Event{}
 				event, hasNext, err := w.next(itr)
 				if err != nil {
-					w.log.V(3).Error(err, "next() failed.")
+					w.log.Error(err, "next() failed.")
 					w.Handler.Error(err)
-					break
+					return
```

Comment:
An (unlikely) error iterating is handled by informing the `Handler`.  An error _should_ end the watch.  Error recovery is the responsibility of the handler which will likely re-create the watch.